iati/events/models.py

```diff
@@ -1,6 +1,249 @@
+        ObjectList(content_panels_en,heading='English content'),
+        ObjectList(content_panels_fr,heading='French content'),
+        ObjectList(content_panels_es,heading='Spanish content'),
+        ObjectList(Page.promote_panels,heading='Promote'),
+        ObjectList(Page.settings_panels,heading='Settings',classname='settings')
+    ])
+    
+    
 
-class Events(Page):
-    pass
\ No newline at end of file
+class EventPage(Page):
+    parent_page_types = ['events.EventIndexPage']
+    subpage_types = []
+    
+    # Start and end dates, needs at least a start date
+    
+    date_start = models.DateTimeField("Event start date and time")
+    date_end = models.DateTimeField("Event end date and time",null=True,blank=True)
+    
+    # Location
+    
+    location = models.TextField(null=True,blank=True)
+    
+    # Registration link
+    
+    registration_link = models.URLField(max_length=255,null=True,blank=True)
+    
+    # Translated page titles, title is English default is go
+    
+    title_fr = models.TextField(null=True,blank=True)
+    title_es = models.TextField(null=True,blank=True)
+    
+    title_translated = TranslatedField(
+        'title',
+        'title_fr',
+        'title_es'
+    )
+    
+    # Heading
+    
+    heading_en = models.TextField(null=True,blank=True)
+    heading_fr = models.TextField(null=True,blank=True)
+    heading_es = models.TextField(null=True,blank=True)
+    
+    heading = TranslatedField(
+        'heading_en',
+        'heading_fr',
+        'heading_es'
+    )
+    
+    # Subheading
+    
+    subheading_en = models.TextField(null=True,blank=True)
+    subheading_fr = models.TextField(null=True,blank=True)
+    subheading_es = models.TextField(null=True,blank=True)
+    
+    subheading = TranslatedField(
+        'subheading_en',
+        'subheading_fr',
+        'subheading_es'
+    )
+    
+    # Description
+    
+    description_en = StreamField(IATIStreamBlock(required=False),null=True,blank=True)
+    description_fr = StreamField(IATIStreamBlock(required=False),null=True,blank=True)
+    description_es = StreamField(IATIStreamBlock(required=False),null=True,blank=True)
+    
+    description = TranslatedField(
+        'description_en',
+        'description_fr',
+        'description_es'
+    )
+    
+    # additional_information
+    
+    additional_information_en = StreamField(IATIStreamBlock(required=False),null=True,blank=True)
+    additional_information_fr = StreamField(IATIStreamBlock(required=False),null=True,blank=True)
+    additional_information_es = StreamField(IATIStreamBlock(required=False),null=True,blank=True)
+    
+    additional_information = TranslatedField(
+        'additional_information_en',
+        'additional_information_fr',
+        'additional_information_es'
+    )
+    
+    event_type = ParentalManyToManyField('events.EventType',blank=True)
+    
+    @property
+    def event_type_concat(self):
+        if translation.get_language() == 'fr':
```

Comment:
I feel like this can be refactored to make it cleaner. When you add the docstring can you add a TODO to it to consider refactoring?